examples/dreamer/dreamer.py

```diff
@@ -0,0 +1,355 @@
+
+                scaler.scale(model_loss_td["loss_world_model"]).backward()
+                scaler.scale(actor_loss_td["loss_actor"]).backward()
+                scaler.scale(value_loss_td["loss_value"]).backward()
+
+                scaler.unscale_(world_model_opt)
+                clip_grad_norm_(world_model.parameters(), cfg.grad_clip)
+                scaler.unscale_(actor_opt)
+                clip_grad_norm_(actor_model.parameters(), cfg.grad_clip)
+                scaler.unscale_(value_opt)
+                clip_grad_norm_(value_model.parameters(), cfg.grad_clip)
+
+                scaler.step(world_model_opt)
+                world_model_opt.zero_grad()
+
+                scaler.step(actor_opt)
+                actor_opt.zero_grad()
+
+                scaler.step(value_opt)
+                value_opt.zero_grad()
+
+                scaler.update()
+
+                with torch.no_grad(), set_exploration_mode("mode"):
+                    td_record = record(None)
+                    if td_record is not None:
+                        for key, value in td_record.items():
+                            if key in ["r_evaluation", "total_r_evaluation"]:
+                                logger.log_scalar(
+                                    key,
+                                    value.detach().cpu().numpy(),
+                                    step=collected_frames,
+                                )
+                    # Compute observation reco
+                    if cfg.record_video and record._count % cfg.record_interval == 0:
+                        true_pixels = recover_pixels(world_model_td["pixels"], stats)
+
+                        reco_pixels = recover_pixels(world_model_td["reco_pixels"], stats)
+                        with autocast(dtype=torch.float16):
+                            world_model_td = world_model_td.select("posterior_states", "next_belief").detach()
+                            world_model_td.batch_size = [
+                                world_model_td.shape[0],
+                                world_model_td.get("next_belief").shape[1],
+                            ]
+                            world_model_td.rename_key("posterior_states", "prior_state")
+                            world_model_td.rename_key("next_belief", "belief")
+                            world_model_td = model_based_env.rollout(
+                                max_steps=true_pixels.shape[1],
+                                policy=actor_model,
+                                auto_reset=False,
+                                tensordict=world_model_td[:, 0],
+                            ).detach()
+                        imagine_pxls = recover_pixels(model_based_env.decode_obs(world_model_td)["reco_pixels"], stats)
+
+                        stacked_pixels = torch.cat([true_pixels, reco_pixels, imagine_pxls], dim=-1)
+                        logger.log_video(
```

Comment:
Why does t appear in the log that we have way more reconstructions than actual pixels? Storing pixels is heavy and can quickly overload the disk

Reply:
"way more reconstructions than actual pixels?" -> What do you mean by this?